src/connector/src/source/kafka/source/reader.rs

```diff
@@ -166,6 +177,10 @@ impl KafkaSplitReader {
                     bytes_current_second = 0;
                     res.clear();
                 }
+                if num_messages >= self.max_num_messages {
+                    yield res;
+                    break 'for_outer_loop;
+                }
```

Comment:
It cannot help control the total number of messages read by rw. Each connector source run in one source executor, what you are doing is limiting the num of messages read by one connector source. 

For example, there are 5 partitions in a kafka topic and 3 source executors in rw. The pr will limit the total message number to `3 * limit`.

Reply:
yes,

that's why 
```
pub bytes_per_second: Option<String>,

    /// This parameter is not intended to be exposed to users.
    /// Since kafka source has no concept of splits and its parallelism
    /// is determined by the compute node's parameter, this parameter specifies
    /// the number of messages total across all the parallelism.
    /// We remark that the default parallelism is the same as the number of Kafka topic partitions
    #[serde(rename = "max.num.messages", alias = "kafka.max.num.messages")]
    pub max_num_messages: Option<String>,
```

The user need to calculate by themselves.

But I thought 
```
We remark that the default parallelism is the same as the number of Kafka topic partitions
```
, so I will remove this comment.

Reply:
changed the comment